react/javascript/demo-embeds/src/hooks/use_listen_embed_events.jsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 Looker Data Sciences, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+import { useRef, useState } from 'react'
+export const useListenEmbedEvents = () => {
+  const [embedEvents, _setEmbedEvents] = useState([])
+  const embedEventsRef = useRef(embedEvents)
+
+  const setEmbedEvents = (events) => {
+    embedEventsRef.current = events
+    _setEmbedEvents(events)
+  }
+
+  const listenEmbedEvents = (embed) => {
+    const eventTypes = [
```

Comment:
Would be great if this was a list we could compose from another collection declared in the embed sdk somewhere. Just sayin'.

Reply:
They are declared as types. As this is javascript (per Tochi's request) not so easy to get at and I'm not sure how easy to iterate at runtime. Having said that, may be it makes sense to add a method that queries helltool as to what events are available. That way it would be version specific and would automatically pick up new events. It would need to be queried after the connection had been established though.

Reply:
Let me know if you'd like to also do a TypeScript version of this. I'd be happy to help.

Reply:
No rush for it as yet. I got one more javascript demo to do (external API). Perhaps after that